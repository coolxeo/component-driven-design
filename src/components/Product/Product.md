Product example:

```js
<Product image="/water-cure.jpg" rate="4.8" title="The Water Cure" subtitle="Joanne Ramos" buttonText="Biography" />
```

Product example without rate & button:

```js
<Product image="/water-cure.jpg" title="The Water Cure" subtitle="Joanne Ramos" />
```