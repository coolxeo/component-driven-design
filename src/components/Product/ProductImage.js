import React from 'react'
import styled from 'styled-components';

const Image = styled.div`
  background-image: url(${props => props.image});
  background-size: cover;
  width: 220px;
  height: 300px;
  border-radius: 18px;
`;

const SmallImage = styled(Image)`
  width: 110px;
  height: 150px;
`

function ProductImage({ image, small }) {
  return (
    <div>
      {!small && <Image image={image} />}
      {small && <SmallImage image={image} />}
    </div>
  )
}

export default ProductImage
