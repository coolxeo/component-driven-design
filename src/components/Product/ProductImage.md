ProductImage example:

```js
<ProductImage image="/water-cure.jpg" />
```

ProductImage Small example:

```js
<ProductImage image="/water-cure.jpg" small={true} />
```