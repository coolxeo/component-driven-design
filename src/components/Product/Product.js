import React from 'react';
import styled from 'styled-components';

import ProductImage from './ProductImage';
import ProductDetails from './ProductDetails/ProductDetails';

const ProductContainer = styled.div`
  display: flex;
  flex-direction: row;
`

function Product({ image, rate, title, subtitle, buttonText }) {
  return (
    <ProductContainer>
      <ProductImage image={image} />
      <ProductDetails rate={rate} title={title} subtitle={subtitle} buttonText={buttonText} />
    </ProductContainer>
  )
}

export default Product;
