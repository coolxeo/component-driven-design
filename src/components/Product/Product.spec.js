import React from 'react'
import renderer from 'react-test-renderer'

import Product from './Product'

it('Product renders correctly', () => {
  const tree = renderer
    .create(
      <Product image="/water-cure.jpg" rate="4.8" title="The Water Cure" subtitle="Joanne Ramos" buttonText="Biography" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});