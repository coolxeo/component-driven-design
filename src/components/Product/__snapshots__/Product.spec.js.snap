// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders correctly 1`] = `
<div
  className="sc-htoDjs lkdQiJ"
>
  <div>
    <div
      className="sc-bdVaJa fmTyxS"
    />
  </div>
  <div
    className="sc-gzVnrw fSNPds"
  >
    <div
      className="sc-htpNat kMQpYq"
    >
      4.8
    </div>
    <div
      className="sc-bxivhb fxQdBO"
    >
      The Water Cure
    </div>
    <div
      className="sc-ifAKCX zuhiC"
    >
      Joanne Ramos
    </div>
    <div>
      <button
        className="sc-EHOje gefPUv"
      >
        Biography
      </button>
    </div>
  </div>
</div>
`;
